ts/src/index.ts

```diff
@@ -0,0 +1,60 @@
+import Discord from 'discord.js'
+import dotenv from "dotenv"
+import { PrismaClient } from "@prisma/client"
+
+dotenv.config()
+const client = new Discord.Client() 
+const prisma = new PrismaClient()
+
+async function repUpdate(userId: number, operation: string): Promise<number> {
+  const exists = await prisma.user.findUnique({where: {id: userId}}) //check if the user already exists 
+  if(!exists) {
+    operation.includes("++") ? // create the user and add rep 0 or 2 depeniing if the contents includes ++
+      await prisma.user.create({data: { id: userId, rep: 2 }}) :
+      await prisma.user.create({data: { id: userId, rep: 0 }})
+    return operation.includes("++") ? 2 : 0 // return rep for message
+  } else {
+    let user = await prisma.user.findUnique({where: {id: userId}}) // get the user
+    operation.includes("++") ?
+      await prisma.user.update({where: {id: userId}, data: { rep: Number(user!.rep + 1) }}) : // update the user
+      await prisma.user.update({where: {id: userId}, data: { rep: Number(user!.rep - 1) }})
+    return operation.includes("++") ?
+      user!.rep + 1 :
+      user!.rep - 1
+  }
+}
+
+client.on('ready', () => {
+  console.log(`Logged in as ${client.user?.tag}!`)
+});
+
+client.on('message', async (msg) => {
+  const args: any[] = msg.content.trim().split(' ') // get args
+  if (
+      msg.content.includes("++") || msg.content.includes("--") // check if the message contains -- or ++
+  ) {
+    let targetName: string = args.join(" ").replace(/(\+\+|--)\s*$/, "") // get the name
+    if(targetName.includes('@')) { // check if the name has a @ in it
+      const pingedUsed: Object = msg.mentions.users.toJSON() // get the mentioned user
+      const userId = Number(pingedUsed[0].id) // get the user id
+      await msg.channel.send(`${pingedUsed[0].username} has rep ${await repUpdate(userId, msg.content)}`)
+      return
+    }
+
+    let target = await msg.guild?.members.fetch() // get all members form the server
+    let user, userName 
+
+    for(const [key, e] of target!.entries()) {
+      if(e.nickname?.includes(targetName)) {user = e; userName = e.nickname; break} // if nickname matches the member break from loop
```

Comment:
I think adding comments above if is more readable

if(bool){ // some comment
}

// some comment
if(bool){ 
}